src/xhydro/frequency_analysis/regional.py

```diff
@@ -0,0 +1,702 @@
+    )
+
+    # Calculating V
+    tau_r_sim = np.dot(n, t_sim) / np.sum(n)
+
+    v_sim = np.sqrt(np.dot(n, (t_sim - tau_r_sim) ** 2) / np.sum(n))
+
+    mu_v = np.mean(v_sim)
+    sigma_v = np.std(v_sim)
+
+    h = (v - mu_v) / sigma_v
+
+    return h, b4, sigma4, tau4_r
+
+
+# Calculating L-moments
+def _momentl_optim(x: np.array) -> list:
+    if x.ndim == 1:
+        x = x[~np.isnan(x)]
+        # reverse sorting
+        x_sort = np.sort(x)
+        x_sort = x_sort[::-1]
+        n = np.size(x_sort)
+        j = np.arange(1, n + 1)
+
+        b0 = np.mean(x_sort)
+        b1 = np.dot((n - j) / (n * (n - 1)), x_sort)
+        b2 = np.dot((n - j) * (n - j - 1) / (n * (n - 1) * (n - 2)), x_sort)
+        b3 = np.dot(
+            (n - j) * (n - j - 1) * (n - j - 2) / (n * (n - 1) * (n - 2) * (n - 3)),
+            x_sort,
+        )
+    elif x.ndim == 2:
+        x.sort()
+        x_sort = np.fliplr(x)
+
+        nn = np.shape(x_sort)
+        j = np.repeat([np.arange(1, nn[1] + 1)], nn[0], axis=0)
+        n = nn[1]
+        b0 = np.mean(x_sort, axis=1)
+        b1 = np.dot((n - j) / (n * (n - 1)), x_sort.T)[0]
+        b2 = np.dot((n - j) * (n - j - 1) / (n * (n - 1) * (n - 2)), x_sort.T)[0]
+        b3 = np.dot(
+            (n - j) * (n - j - 1) * (n - j - 2) / (n * (n - 1) * (n - 2) * (n - 3)),
+            x_sort.T,
+        )[0]
+    else:
+        raise NotImplementedError("Only 1d and 2d have been implemented")
+
+    # Moment L
+    lambda1 = b0
+    lambda2 = 2 * b1 - b0
+    lambda3 = 6 * b2 - 6 * b1 + b0
+    lambda4 = 20 * b3 - 30 * b2 + 12 * b1 - b0
+
+    tau = lambda2 / lambda1  # L_CV # tau2 in Anctil, tau dans Hosking et al.
+    tau3 = lambda3 / lambda2  # L_CS
+    tau4 = lambda4 / lambda2  # L_CK
+
+    return [lambda1, lambda2, lambda3, tau, tau3, tau4]
+
+
+def _append_ds_vars_names(ds: xr.Dataset, suffix: str) -> xr.Dataset:
+    for name in ds.data_vars:
+        ds = ds.rename({name: name + suffix})
+    return ds
+
+
+def mask_h_z(
+    ds: xr.Dataset, thresh_h: float | None = 1, thresh_z: float | None = 1.64
+) -> xr.DataArray:
+    """
+    Create a boolean mask based on heterogeneity measure H and Z-score thresholds.
+
+    Parameters
+    ----------
+    ds : xarray.Dataset
+        Dataset containing H and Z values for each group.
+    thresh_h : float, optional
+        Threshold for the heterogeneity measure H. Default is 1.
+    thresh_z : float, optional
+        Threshold for the absolute Z-score. Default is 1.64.
+
+    Returns
+    -------
+    xarray.DataArray
+        Boolean mask where True indicates groups that meet both threshold criteria.
+    """
+    ds_out = (ds.sel(crit="H") < thresh_h) & (abs(ds.sel(crit="Z")) < thresh_z)
+    for v in ds_out.var():
+        ds_out[v].attrs["H_threshold"] = thresh_h
+        ds_out[v].attrs["Z_threshold"] = thresh_z
+        ds_out[v].attrs["description"] = "Mask for regions based on H & Z thresholds"
```

Comment:
C'est toujours bien d'avoir un `long_name`.
```suggestion
        ds_out[v].attrs["Z_threshold"] = thresh_z
        ds_out[v].attrs["long_name"] = "Mask"
        ds_out[v].attrs["description"] = "Mask for regions based on H & Z thresholds"
```